### DISCUSSION TITLE
Importing app into a staging container that has plugins

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#11765] [Help] [answered]
Hi,

We are trying to minimize risk on production apps by setting up a second instance of budibase where we can run a published app for testing.

Our setup is as follows:

1. Main instance:  single image podman pod of BB for development and production - this one is licensed
2. Testing instance: single image podman pod of BB for testing

We develop the apps in the main instance and publish them as production ready on that instance. We have found that:

- The inability to test workflows within the context of a user interacting with an app in the "preview" state adds a lot of risk. 
- The inability to share a link to the preview state with end users to perform UAT also adds risk.

We are trying to work around this by exporting the development app and importing it into the test pod but have found that any plugins in the exported app are removed when importing, despite the plugins existing in that instance.

Is there any way to work around this at the moment?

Thanks

**Comment by benaunt** [comment, marked as answer]
Oh we just realised, if we re-add just one plugin to a page, all plugins suddenly re-appear through the whole app with their configuration correct.

**Reply by benaunt** [reply]
Spoke to early, only some plugins appeared, not all :(

**Reply by melohagan** [reply]
Hey @benaunt 

You need to add one of each plugin you're using. I've raised an issue here: https://github.com/Budibase/budibase/issues/9260
